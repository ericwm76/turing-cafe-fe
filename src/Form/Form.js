import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      guests: ''
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  updateState = (e) => {
    e.preventDefault();
    if (this.state.name && this.state.date && this.state.time && this.state.guests) {
      this.props.createNewReservation(this.state.name, this.state.date, this.state.time, this.state.guests)
    }
  }

  render() {
    return (
      <div>
        <input className='name-input' name='name' value={this.state.name} type='text' placeholder='Name' onChange={this.handleChange}/>
        <input className='date-input' name='date' value={this.state.date} type='date' placeholder='Date' onChange={this.handleChange}/>
        <input className='time-input' name='time' value={this.state.time} type='text' placeholder='Time' onChange={this.handleChange}/>
        <input className='guests-input' name='guests' value={this.state.guests} type='number' placeholder='Number of guests' onChange={this.handleChange}/>
        <button onClick={this.updateState}>Make reservation</button>
      </div>
    )
  }
}

export default Form;