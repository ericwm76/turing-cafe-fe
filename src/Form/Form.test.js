import React from 'react';
import { shallow } from 'enzyme'
import Form from './Form';

describe('Form', () => {
  let wrapper, mockEvent;
  let createNewReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form createNewReservation={createNewReservation} />);
    mockEvent = {
      preventDefault: jest.fn(),
      target: {
        name: 'name',
        value: 'MK'
      }
    }
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleChange on change in the input field', () => {
    wrapper.instance().handleChange = jest.fn();
    wrapper.find('input').at(0).simulate('change', mockEvent);
    expect(wrapper.instance().handleChange).toHaveBeenCalled();
  })

  it('should update state when handleChange is called', () => {
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('name')).toEqual('MK')
  });

  it('should call updateState when Make Reservation button is clicked', () => {
    wrapper.instance().updateState = jest.fn();
    wrapper.setState({
      name: 'Hey',
      date: 'Today',
      time: 'Now',
      guests: '30'
    });

    wrapper.find('button').simulate('click');
    expect(wrapper.instance().updateState).toHaveBeenCalled()
  })

  it('should call createNewReservation when updateState is called', () => {
    wrapper.setState({
      name: 'Hey',
      date: 'Today',
      time: 'Now',
      guests: '30'
    });

    wrapper.instance().updateState(mockEvent);
    expect(createNewReservation).toHaveBeenCalled();


  })

})