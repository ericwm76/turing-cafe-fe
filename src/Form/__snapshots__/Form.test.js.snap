// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<div>
  <input
    className="name-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="date-input"
    name="date"
    onChange={[Function]}
    placeholder="Date"
    type="date"
    value=""
  />
  <input
    className="time-input"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    className="guests-input"
    name="guests"
    onChange={[Function]}
    placeholder="Number of guests"
    type="number"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Make reservation
  </button>
</div>
`;
